jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java

```diff
@@ -751,6 +753,14 @@ public String getCorrelationKey() {
     public void setCorrelationKey(String correlationKey) {
         this.correlationKey = correlationKey;
     }
+    
+    public Date getStartDate() {
+        return startDate;
+    }
+    
+    public void internalSetStartDate(Date startDate) {
+        this.startDate = startDate;
+    }
```

Comment:
"in addition allow setting it only when it was not set yet." option is still missing.

Reply:
Done